libs/common/src/System_Win32.cpp

```diff
@@ -125,14 +129,14 @@ std::string System::getUserName()
 {
     DWORD nameLen = 0;
     // Query required size
-    GetUserNameW(nullptr, &nameLen);
+    GetUserNameExW(EXTENDED_NAME_FORMAT::NameDisplay, nullptr, &nameLen); // nameLen contains terminating 0 here
     if(nameLen == 0)
         throw std::runtime_error("Could not query username length");
     std::vector<wchar_t> userName(nameLen);
-    if(GetUserNameW(userName.data(), &nameLen) == 0)
+    if(GetUserNameExW(EXTENDED_NAME_FORMAT::NameDisplay, userName.data(), &nameLen) == 0)
         throw std::runtime_error("Could not get username");
 
-    userName.resize(nameLen - 1); // nameLen already contains terminating 0
+    userName.resize(nameLen); // nameLen doesn't contain terminating 0 here
```

Comment:
This doesn't seem right to me. In line 132 you write it does contain the terminating zero but here that it doesn't. Can you explain this maybe with a reference or revert this line please?

Reply:
After revisiting the docs for `GetUserNameExW()`, I see that what I meant is that first call to it returns the length with null, as the input is nullptr+size0, but the second call returns the length without null.

I'll improve comments after I'm done with the work, since they indeed are imprecise/misleading.

Reply:
Found on https://learn.microsoft.com/en-us/windows/win32/api/secext/nf-secext-getusernameexw

>  If the function is successful, the variable receives the number of TCHARs copied to the buffer, not including the terminating null character.

> If lpNameBuffer is too small, the function fails [...]This parameter receives the required buffer size, in Unicode characters (whether or not Unicode is being used), including the terminating null character.

Premium design... :-D